contrib/cirrus/lib.sh

```diff
@@ -238,34 +238,44 @@ ircmsg() {
 # there is at least one release tag not having any '-' characters (return 0)
 # or otherwise (return non-0).
 is_release() {
-    req_env_var CIRRUS_BASE_SHA CIRRUS_CHANGE_IN_REPO
-    local range="${CIRRUS_BASE_SHA}..${CIRRUS_CHANGE_IN_REPO}"
-    # Easy check first, default non-useful values
-    if echo "${range}$CIRRUS_TAG" | grep -iq 'unknown'; then
-        die 11 "is_release() unusable range ${range} or tag $CIRRUS_TAG"
-    fi
-    # Next easy check, is CIRRUS_TAG set
     unset RELVER
+    local ret
+    req_env_var CIRRUS_CHANGE_IN_REPO
     if [[ -n "$CIRRUS_TAG" ]]; then
         RELVER="$CIRRUS_TAG"
-    else # Lastly, look through the range for tags
-        git fetch --all --tags &> /dev/null|| \
-            die 12 "is_release() failed to fetch tags"
-        RELVER=$(git log --pretty='format:%d' $range | \
-                 grep '(tag:' | sed -r -e 's/\s+[(]tag:\s+(v[0-9].*)[)]/\1/' | \
-                 sort -uV | tail -1)
+    elif [[ ! "$CIRRUS_BASE_SHA" =~ "unknown" ]]
+    then
+        req_env_var CIRRUS_BASE_SHA
```

Comment:
I think the 'undef or empty' case is only possible in the unit-tests.  Otherwise, when loading `lib.sh` it is guaranteed non-empty (though it may be set to `unknown123456789`).  I should add a comment to this line, it is only here for unit-testing.